### PR TITLE
Manage dashboards as code

### PR DESCRIPTION
**Pull request by vmercierfr**
# Objective

Manage dashboards as code

# Why

Dashboards are currently manually edited via Grafana web UI and published on Grafana labs. This presents us with maintenance difficulties:
- End-user may use Prometheus RDS exporter version that is incompatible with dashboards
- End-users are not aware of dashboards updates
- Versioning in Grafana labs does not reflect the dashboard version
- Difficulites to have consistency betweens panels of all our dashboards

In this PR we are introducing dashboard as code, using [Grafonnet](https://grafana.github.io/grafonnet/index.html). Which is a Jsonnet library maintainted by Grafana that allow generating all Grafana resources.

By defining dashboards as code, I took the opportunity to fix incorrect Prometheus queries and make sure each panel has a description.

As the release cycle on grafana.com doesn't provide a good experience, we're going to quickly offer to deploy dashboards via the [Grafana operator](https://grafana.github.io/grafana-operator/docs/dashboards/). This will eventually enable us to deploy the export and dashboards in a single deployment.

On long term, Grafonnet librairies could be moved to an external repositories to have consistency with upcoming PostgreSQL dashboards of [DMF](https://github.com/qonto/database-monitoring-framework).

# How

- Add dashboards source (managed by Grafonnet)
- Add precommit hooks to detect uncommit dashboards changes

# Release plan

- [ ] Merge this PR
- [ ] Apply with CI
- [ ] Publish new dashboards

### CODE REVIEW

**Comment by vmercierfr** [comment]
#sre

**Comment by vmercierfr** [comment]
cc @ortizjavier you could be interested by this dashboard as code feature

**Comment by airclovis** [comment]
Should we explain how to test the rendered dashboards locally using the docker-compose file in `scripts/prometheus`? I don't know if we can change the volumes to automatically load the generated dashboards in `configs/grafana/public` or if we need to manually import them as `json` from the UI.

**Comment by vmercierfr** [comment]
> Should we explain how to test the rendered dashboards locally using the docker-compose file in `scripts/prometheus`? I don't know if we can change the volumes to automatically load the generated dashboards in `configs/grafana/public` or if we need to manually import them as `json` from the UI.
> https://github.com/qonto/prometheus-rds-exporter/pull/133#issuecomment-1993728950

The [local development environment](https://github.com/qonto/prometheus-rds-exporter?tab=readme-ov-file#development-environment) already start everything out of the box, including dashboard. And the [grafana-dashboard-update.md](https://github.com/qonto/prometheus-rds-exporter/blob/grafonnet/docs/grafana-dashboard-update.md) explains how to update dashboards.

I updated the docs and configuration to be more explicit.
@airclovis does it fit your expectations?
Feel free to start the local development, test a dashboard and share improvements.

**Review by airclovis — APPROVED** [review]
Lgtm 🙂

File: scripts/prometheus/configuration/grafana/entrypoint.sh
Comment (qfritz) [inline scripts/prometheus/configuration/grafana/entrypoint.sh:1]: replace PROVISIONNING -> PROVISIONING (lose the double N)

**Review by qfritz — APPROVED** [review]
I don't have time to do a deeper review but for now this looks good. Remarks may arise when I use your components as a basis to build SQL Exporter dashboards.

**Review by dcupif — APPROVED** [review]
lgtm

**Pull request merged**